pkg/json_probe_results.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkg
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+)
+
+// JSONScorecardProbeResult exports results as JSON for flat findings without checks.
+//
+//nolint:govet
+type JSONScorecardProbeResult struct {
+	Date      string            `json:"date"`
+	Repo      jsonRepoV2        `json:"repo"`
+	Scorecard jsonScorecardV2   `json:"scorecard"`
+	Findings  []finding.Finding `json:"findings"`
+	Metadata  map[string]string `json:"metadata"`
+}
+
+// TODO: finsinds should enventually be part of the scorecard structure.
+func (r *ScorecardResult) AsPJSON(writer io.Writer,
+) error {
+	encoder := json.NewEncoder(writer)
+	out := JSONScorecardProbeResult{
+		Repo: jsonRepoV2{
+			Name:   r.Repo.Name,
+			Commit: r.Repo.CommitSHA,
+		},
+		Scorecard: jsonScorecardV2{
+			Version: r.Scorecard.Version,
+			Commit:  r.Scorecard.CommitSHA,
+		},
+		Date:     r.Date.Format("2006-01-02"),
+		Findings: r.ProbeResults,
+	}
```

Comment:
I did not just because 
- I wanted some feedback o nest vs flat structure (one of your earlier comment). 
- I don't know if this is needed for practical purposes, so I did not want to over-commit to it right away

I added it back. It will show as:
```json
"metadata": {
    "repository.defaultBranch": "main",
    "repository.host": "github.com",
    "repository.name": "ossf/scorecard-action",
    "repository.sha1": "7fe8d54f4acf26efc826129c396173562e2c766c",
    "repository.uri": "github.com/ossf/scorecard-action"
  }
```

Let me know if you'd like me to keep it or not. If not, I'll create a PR for tracking so that we can re-visit

Reply:
If you don't want to over-commit right now I would remove the `Metadata` field from `JSONScorecardProbeResult`

Reply:
ok, rolling back then.